extension/js/common/platform/catch.ts

```diff
@@ -123,15 +123,23 @@ export class Catch {
       && typeof (v as Promise<unknown>).catch === 'function'; // tslint:disable-line:no-unbound-method - only testing if exists
   };
 
-  public static try = (code: () => void | Promise<void>) => {
+  public static try = (code: () => void | Promise<void>, callback?: () => void | Promise<void>) => {
     return () => { // returns a function
+      let isPromise = false;
       try {
         const r = code();
         if (Catch.isPromise(r)) {
+          isPromise = true;
           r.catch(Catch.reportErr);
+          void r.finally(callback);
         }
       } catch (codeErr) {
         Catch.reportErr(codeErr);
+      } finally {
+        // Do not call callback for promise function because it's already called in line 134.
+        if (!isPromise && callback) {
+          callback();
```

Comment:
Don't we need to `catch` this too? This looks excessively complicated.
Why don't we add a plain `try .. catch .. finally` around lines 301 and 302?
```
try {
         const { privateKeys } = await keyManager.getPrivateKeys(idToken);
         await processKeysFromEkm(acctEmail, privateKeys.map(entry => entry.decryptedPrivateKey), clientConfiguration, factory, idToken, ppEvent);
} finally {
         await comlpetion();
}
```